### PR TITLE
Add support for parenthesized expressions & EXISTS subquery

### PR DESCRIPTION
**Pull request by DerekStride**
## What

There's three changes being made in this PR, each are contained to individual commits.
1. Support for parenthesized expression. e.g. wrapping expressions in `(` & `)` to group them.
2. Support for the [`EXISTS`](https://dev.mysql.com/doc/refman/8.0/en/exists-and-not-exists-subqueries.html) expression.
    - Includes subquery support for `IN`
    - Excludes `NOT IN`, there was conflicts with precedence that I want to sort out separately.
3. Support for [`ANY`, `SOME`, & `ALL`](https://www.postgresql.org/docs/current/functions-subquery.html)

### CODE REVIEW

**Comment by DerekStride** [comment]
FYI I've enabled branch protection to require at least 1 approval & all the test suites passing before merging a PR

File: grammar.js
Comment (matthias-Q) [inline grammar.js:1485]: Postgres seems to extend this syntax a bit https://www.postgresql.org/docs/current/functions-subquery.html

Does it make sense to add `not in`, `any`, `some`... as well?
`in` does work already, but only for lists and not for subqueries.

**Review by matthias-Q — COMMENTED** [review]
Parenthesis change LGTM
`exists` is good, but adding more options (like `not in` etc) would be great

**Comment by DerekStride** [comment]
I've update the PR (& the PR description) to include the suggested changes.

The one exception was `NOT IN` ended up causing precedence issues. I can open a separate issue to explore that in the future.

**Comment by matthias-Q** [comment]
LGTM

**Review by matthias-Q — APPROVED** [review]

**Pull request merged**
